metricbeat/module/system/service/service.go

```diff
@@ -125,17 +162,68 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		if !isOpen {
 			return nil
 		}
+	}
+
+	err = m.reportInactiveInst(report)
+	if err != nil {
+		return errors.Wrap(err, "error reporting inactive services")
+	}
+
+	return nil
+}
+
+func (m *MetricSet) reportInactiveInst(r mb.ReporterV2) error {
+	for name, inst := range m.instanceTracker {
+		fmt.Printf("====checking unit: %s\n", name)
+		// If the unit has timed out, remove it
+		if time.Now().Sub(inst.lastReported) >= m.cfg.InstantiatedTimeout {
```

Comment:
Instead of removing it after a timeout, could we check with `systemctl status` (or `getProps` or its api equivalent) to check if the unit is still available, and remove it only when it isn't? systemd seems to be keeping track of these units after its process dies.
```
root@old-systemd:~# systemd --version
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN
root@old-systemd:~# systemctl status tester@10.service
● tester@10.service - Instance tester time 10
   Loaded: loaded (/lib/systemd/system/tester@.service; static; vendor preset: enabled)
   Active: inactive (dead)

May 28 12:26:40 old-systemd systemd[1]: Started Instance tester time 10.
```

Reply:
@jsoriano what OS is that? I can't seem to reproduce that timer, and I can't see anything in the underlying dbus API.
```
vagrant@ubuntu-xenial:~$ systemd --version
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN
vagrant@ubuntu-xenial:~$ sudo systemctl start tester@10.service
vagrant@ubuntu-xenial:~$ systemctl status tester@10.service
● tester@10.service - Instance tester time 10
   Loaded: loaded (/lib/systemd/system/tester@.service; static; vendor preset: enabled)
   Active: inactive (dead)
vagrant@ubuntu-xenial:~$
```